crates/solidity/inputs/schema/grammar/topics/04-statements/02-declaration-statements.yml

```diff
@@ -12,10 +12,16 @@
                 separator: ","
                 expression:
                   optional:
-                    sequence:
-                      - optional:
-                          reference: "TypeName"
-                      - reference: "Identifier"
+                    choice:
```

Comment:
Tried fixing using a `lookahead`, but turns out that is not usable on rules. Open to suggestions if there is a better way to produce this. I don't like the fact that the CST shape is different between the two choices, but we can revisit this later as well.

Reply:
Also a temporary fix until we investigate later. The above won't work for the case of `foo`. It will be parsed as a `TypeName -> IdentifierPath` , then it will fail later for a missing `Identifier`. Instead of parsing `foo` directly as the last `Identifier`.

Reply:
Annoying!

Reply:
@AntonyBlakey I tried something like the following, but parser codegen doesn't generate the lookahead:

```yaml
                    sequence:
                      - optional:
                          reference: "TypeName"
                          lookahead:
                            choice:
                              - reference: "DataLocation"
                              - reference: "Identifier"
                      - optional:
                          reference: "DataLocation"
                      - reference: "Identifier"
```